src/parser/statement.js

```diff
@@ -940,6 +938,22 @@ pp.checkExport = function (node, checkNames, isDefault) {
   }
 };
 
+pp.checkDeclaration = function(node, isDefault) {
```

Comment:
We actually might not need to pass through `isDefault` here, since `default` is a reserved word and can't be the name for an `Identifier`